iroh-net/src/relay/client.rs

```diff
@@ -1,511 +1,1112 @@
-//! based on tailscale/derp/derp_client.go
-use std::net::SocketAddr;
-use std::pin::Pin;
+//! Based on tailscale/derp/derphttp/derphttp_client.go
```

Comment:
As you can see, the diff here is fucked :grimacing: 

I moved `relay/http/client.rs` to here, and moved the old `relay/client.rs` to `relay/client/conn.rs`.
I also renamed a bunch of the `Client*` things to `Conn*`, which probably made this diff look bad.